From 9b23f0942ce186d31df008ff33a4193b8e498500 Mon Sep 17 00:00:00 2001
From: Huang Shan <huang.shan@freescale.com>
Date: Thu, 8 May 2014 13:20:52 +0800
Subject: [PATCH 65/70] ENGR00303833-2 Ethernet: change cancel action in
 advance dialog

Override the function of show to read the orignal proxy information
And fix the issue of ExclusionList being set null

Signed-off-by: Huang Shan <huang.shan@freescale.com>
---
 Ethernet/com/fsl/ethernet/EthernetAdvDialog.java | 10 ++++++++--
 Ethernet/com/fsl/ethernet/EthernetManager.java   |  3 +++
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
index d0b7728..8784aff 100644
--- a/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
+++ b/Ethernet/com/fsl/ethernet/EthernetAdvDialog.java
@@ -77,11 +77,10 @@ public class EthernetAdvDialog extends AlertDialog implements DialogInterface.On
 
     public void handle_saveconf() {
         EthernetDevInfo info = new EthernetDevInfo();
-
         if ((mProxyIp.getText().toString().equals(""))||(mProxyPort.getText().toString().equals("")))
         {
-            Toast.makeText(this.getContext(), R.string.advanced_need_setting,Toast.LENGTH_SHORT).show();
         }else{
+            info.setIfName("eth0");
             info.setConnectMode(mEthEnabler.getManager().getSharedPreMode());
             info.setIpAddress(mEthEnabler.getManager().getSharedPreIpAddress());
             info.setDnsAddr(mEthEnabler.getManager().getSharedPreDnsAddress());
@@ -113,4 +112,11 @@ public class EthernetAdvDialog extends AlertDialog implements DialogInterface.On
     }
 
 
+  @Override
+    public void show() {
+          super.show();
+          mProxyIp.setText(mEthEnabler.getManager().getSharedPreProxyAddress(),TextView.BufferType.EDITABLE);
+          mProxyPort.setText(mEthEnabler.getManager().getSharedPreProxyPort(),TextView.BufferType.EDITABLE);
+          mProxyExclusionList.setText(mEthEnabler.getManager().getSharedPreProxyExclusionList(),TextView.BufferType.EDITABLE);
+    }
 }
diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index bde153c..fac1edd 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -147,6 +147,7 @@ public class EthernetManager {
         info.setDnsAddr(getSharedPreDnsAddress());
         info.setProxyAddr(getSharedPreProxyAddress());
         info.setProxyPort(getSharedPreProxyPort());
+        info.setProxyExclusionList(getSharedPreProxyExclusionList());
         return info;
     }
 
@@ -233,8 +234,10 @@ public class EthernetManager {
         infotemp.setDnsAddr(dns);// now only use dns1, need optimization later here.
         String proxyAddress = getSharedPreProxyAddress();
         String proxyPort = getSharedPreProxyPort();
+        String proxyExclusionList=getSharedPreProxyExclusionList();
         infotemp.setProxyAddr(proxyAddress);
         infotemp.setProxyPort(proxyPort);
+        infotemp.setProxyExclusionList(proxyExclusionList);
         return infotemp;
     }
 
-- 
1.8.0

